core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1402,9 +1402,10 @@ class KafkaApis(val requestChannel: RequestChannel,
       sendResponseMaybeThrottle(request, createResponse)
     }
 
-    if (!controller.isActive) {
+    if (!controller.isActive || !config.deleteTopicEnable) {
+      val error = if (!controller.isActive) Errors.NOT_CONTROLLER else Errors.INCOMPATIBLE_WITH_CONFIG
```

Comment:
I don't think we can return a whole new error code without an explicit change in the binary format (think about what happens in a 11.0.0 client gets this new error code from a 11.0.2 broker, for example). For old clients we could use an existing error code. `POLICY_VIOLATION` has similar semantics, so maybe that would be a good choice, though @ijuma would be better placed to advise about this.

Reply:
Though thinking about it, this API doesn't currently return `POLICY_VIOLATION`, so even that would probably need a new binary version (@ijuma?). In which case it may be better to fix in as part of KIP-201.

Reply:
Yes, we should try to reuse an error code that is already used by this protocol API.

Reply:
Which one do you think fits better? ``INVALID_REQUEST``?

Reply:
I used ``INVALID_REQUEST``, ok?